torch/csrc/autograd/function.h

```diff
@@ -97,6 +107,13 @@ struct TORCH_API Node : std::enable_shared_from_this<Node> {
       next_edges_(std::move(next_edges)) {
     if (AnomalyMode::is_enabled()) {
       metadata()->store_stack();
+
+      // If anomaly mode is enabled and graph is constructed, then assign this
+      // node to a parent. A parent is a Node where this Node is created.
+      // We are tracking the parents to track multiple backward operations.
+      if (GradMode::is_enabled()) {
```

Comment:
Is this check really necessary?
We want to save this (if the thread local exists) all the time no?

Reply:
Yes, there is no drawback in saving the parent all the time in anomaly mode.